src/spec.c

```diff
@@ -487,6 +488,13 @@ IndexSpec *IndexSpec_CreateNew(RedisModuleCtx *ctx, RedisModuleString **argv, in
     IndexSpec_SetTimeoutTimer(sp, StrongRef_Demote(spec_ref));
   }
 
+  // (Lazily) Subscribe to keyspace notifications, now that we have at least one
+  // spec
+  if (!RS_KeyspaceEvents_Initialized) {
+    Initialize_KeyspaceNotifications(ctx);
+    RS_KeyspaceEvents_Initialized = true;
+  }
```

Comment:
Why not moving this logic into `Initialize_ServerEventNotifications` instead of writing it 3 times?

Reply:
`Initialize_ServerEventNotifications` is called on server load where we don't want to subscribe to these notifications.
I can gather these lines to a function if that's what you mean